src/index.ts

```diff
@@ -372,3 +376,53 @@ export class CachedClient implements ICachedHttpClient {
     return await this._httpClient.delete(route);
   }
 }
+
+export const redact = (options: RequestOptions) => {
+  const clone = copy(options);
```

Comment:
Felt safest to take a deep copy of this, since I don't know what kind of asynchronous access `got` might have on this after the fact.